File: Data Structures & Algorithms/longest-common-prefix/submission-5.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        string res = strs[0];
        for (int i=1;i<strs.size();i++) {
            string curr = "";
            string word = strs[i];
            for (int j=0;j<word.length();j++) {
                if (j == res.length() || word[j] != res[j]) {
                    break;
                } else {
                    curr += word[j];
                }
            }
            if (curr.length() < res.length()) res = curr;
        }
        return res;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-5.py
from collections import defaultdict

def my_sort(word: str) -> str:
    letters = list(word)
    letters.sort()
    return "".join(letters)

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = defaultdict(list)
        for word in strs:
            res[my_sort(word)].append(word)
        return list(res.values())